singularity-eos/eos/eos_vinet.hpp

```diff
@@ -0,0 +1,374 @@
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::BulkModulusFromDensityTemperature(
+    const Real rho, const Real temp, Real *lambda) const {
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[7] * output[7] * rho;
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::TemperatureFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real Tref;
+  Real output[8];
+  Tref = _T0;
+  Vinet_F_DT_func(rho, Tref, output);
+  return (sie - output[0]) / _Cv0 + Tref;
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::PressureFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real temp;
+  Real output[8];
+  temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  Vinet_F_DT_func(rho, temp, output);
+  return output[1];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::EntropyFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real temp;
+  Real output[8];
+  temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  Vinet_F_DT_func(rho, temp, output);
+  return output[6];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::BulkModulusFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real temp;
+  Real output[8];
+  temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  Vinet_F_DT_func(rho, temp, output);
+  return output[7] * output[7] * rho;
+}
+// AEM: Give error since function is not well defined
+PORTABLE_INLINE_FUNCTION void
+Vinet::DensityEnergyFromPressureTemperature(const Real press, const Real temp,
+                                            Real *lambda, Real &rho, Real &sie) const {
+  EOS_ERROR("Vinet::DensityEnergyFromPressureTemperature: "
+            "Not implemented.\n");
+}
+// AEM: Left this in since I am not sure what it does.
+PORTABLE_INLINE_FUNCTION void Vinet::FillEos(Real &rho, Real &temp, Real &sie,
+                                             Real &press, Real &cv, Real &bmod,
+                                             const unsigned long output,
+                                             Real *lambda) const {
+  // The following could be sped up with work!
+  const unsigned long input = ~output;
+  if (thermalqs::temperature & input && thermalqs::pressure & input) {
+    DensityEnergyFromPressureTemperature(press, temp, lambda, rho, sie);
+  } else if (thermalqs::density & output ||
+             thermalqs::specific_internal_energy & output) {
+    EOS_ERROR("Vinet FillEos: Density and energy are currently required inputs "
+              "except when pressure and temperature are inputs.\n");
+  }
+  if (output & thermalqs::pressure) press = PressureFromDensityInternalEnergy(rho, sie);
+  if (output & thermalqs::temperature)
+    temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  if (output & thermalqs::bulk_modulus)
+    bmod = BulkModulusFromDensityInternalEnergy(rho, sie);
+  if (output & thermalqs::specific_heat)
+    cv = SpecificHeatFromDensityInternalEnergy(rho, sie);
+}
```

Comment:
This function is important and should be revisited if the `Vinet` eos is going to be used outside of tests. The idea here is that the user requests what variables they want and those are filled. The EOS fills them in the order it believes is most appropriate based its internal representation. There are significant opportunities for a performance boost here, especially with your `Vinet_F_DT_func` since you compute everything always anyway.